packages/xdl/src/project/createBundlesAsync.ts

```diff
@@ -24,17 +24,59 @@ type PackagerOptions = {
   minify: boolean;
 };
 
-export function printBundleSizes(bundles: { android: BundleOutput; ios: BundleOutput }) {
-  const files = [
-    ['index.ios.js', bundles.ios.code],
-    ['index.android.js', bundles.android.code],
-  ];
-  // Account for inline source maps
-  if (bundles.ios.map) {
-    files.push([chalk.dim('index.ios.js.map'), bundles.ios.map]);
-  }
-  if (bundles.android.map) {
-    files.push([chalk.dim('index.android.js.map'), bundles.android.map]);
+// If `isHermesPreferable` is true, will try use hermes bytecode bundle as primary output.
+// This if for classic `expo export` or `expo publish` flows.
+export function printBundleSizes(
+  bundles: { android: BundleOutput; ios: BundleOutput },
+  isHermesPreferable: boolean
+) {
+  const files: [string, string | Uint8Array][] = [];
+
+  if (isHermesPreferable) {
+    if (bundles.ios.hermesBytecodeBundle) {
+      files.push(['index.ios.js (Hermes)', bundles.ios.hermesBytecodeBundle]);
+    } else {
+      files.push(['index.ios.js', bundles.ios.code]);
+    }
+    if (bundles.android.hermesBytecodeBundle) {
+      files.push(['index.android.js (Hermes)', bundles.android.hermesBytecodeBundle]);
+    } else {
+      files.push(['index.android.js', bundles.android.code]);
+    }
+
+    // Account for inline source maps
+    if (bundles.ios.hermesSourcemap) {
+      files.push([chalk.dim('index.ios.js.map (Hermes)'), bundles.ios.hermesSourcemap]);
+    } else if (bundles.ios.map) {
+      files.push([chalk.dim('index.ios.js.map'), bundles.ios.map]);
+    }
+    if (bundles.android.hermesSourcemap) {
+      files.push([chalk.dim('index.android.js.map (Hermes)'), bundles.android.hermesSourcemap]);
+    } else if (bundles.android.map) {
+      files.push([chalk.dim('index.android.js.map'), bundles.android.map]);
+    }
+  } else {
+    files.push(['index.ios.js', bundles.ios.code], ['index.android.js', bundles.android.code]);
+    if (bundles.ios.hermesBytecodeBundle) {
+      files.push(['index.ios.js (Hermes)', bundles.ios.hermesBytecodeBundle]);
+    }
```

Comment:
Why are we pushing a `hermes` bundle here if `isHermesPreferable === false`?

Reply:
this is for eas update. saying that we will upload both the plaintext js bundle and hermes bytecode bundle to our servers.
is the design making sense?

personally, i am not sure if we need to do this in the meantime.
some benefit foreseeing is for dev-clients to load plaintext javascript bundle if hermes bytecode version is incompatible.
cc @brentvatne @ide 

if we don't need design, the code here could be simpler.

Reply:
I believe the last time we (@brentvatne @ide @esamelson) discussed this, we were going to avoid fallback to js due to the complexity and leave it to the client to bump their runtime version correctly. Would be curious to know if I misunderstood this though.

Reply:
just revisited our discussion and yes you are right.
sorry for getting lost from the beginning.